sql: missing retry loop in TestSchemaChangeAfterCreateInTxn

SHA: https://github.com/cockroachdb/cockroach/commits/759e44503290a50fba412804bb4432256757fbed

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestSchemaChangeAfterCreateInTxn PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=918319&tab=buildLog

```
I180921 05:46:44.275397 140926 storage/split_queue.go:116  [n1,split,s1,r2/1:/System/{-NodeLive…}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275424 140926 storage/split_queue.go:116  [n1,split,s1,r19/1:/Table/2{2-3}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275452 140926 storage/split_queue.go:116  [n1,split,s1,r11/1:/Table/1{4-5}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275485 140926 storage/split_queue.go:116  [n1,split,s1,r14/1:/Table/1{7-8}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275512 140926 storage/split_queue.go:116  [n1,split,s1,r20/1:/Table/{23-50}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275538 140926 storage/split_queue.go:116  [n1,split,s1,r9/1:/Table/1{2-3}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275569 140926 storage/split_queue.go:116  [n1,split,s1,r16/1:/Table/{19-20}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275595 140926 storage/split_queue.go:116  [n1,split,s1,r18/1:/Table/2{1-2}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275617 140926 storage/split_queue.go:116  [n1,split,s1,r6/1:/{System/tse-Table/System…}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275646 140926 storage/split_queue.go:116  [n1,split,s1,r7/1:/Table/{SystemCon…-11}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275673 140926 storage/split_queue.go:116  [n1,split,s1,r8/1:/Table/1{1-2}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275699 140926 storage/split_queue.go:116  [n1,split,s1,r10/1:/Table/1{3-4}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275731 140926 storage/split_queue.go:116  [n1,split,s1,r13/1:/Table/1{6-7}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275758 140926 storage/split_queue.go:116  [n1,split,s1,r12/1:/Table/1{5-6}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.275853 141326 storage/replica_command.go:298  [n1,split,s1,r22/1:/{Table/51-Max}] initiating a split of this range at key /Table/52 [r23]
I180921 05:46:44.280752 140863 storage/split_queue.go:116  [n1,split,s1,r22/1:/Table/5{1-2}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.280779 140863 storage/split_queue.go:116  [n1,split,s1,r23/1:/{Table/52-Max}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.281651 141326 storage/split_queue.go:170  [n1,split,s1,r22/1:/Table/5{1-2}] split done
I180921 05:46:44.863143 140972 storage/split_queue.go:116  [n1,split,s1,r1/1:/{Min-System/}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:44.877845 141040 rpc/nodedialer/nodedialer.go:92  [n1,consistencyChecker,s1,r1/1:/{Min-System/}] connection to n1 established
I180921 05:46:45.856006 140777 gossip/gossip.go:939  [n1] gossip connectivity
  n1 [sentinel];
I180921 05:46:45.863493 140972 storage/split_queue.go:116  [n1,split,s1,r20/1:/Table/{23-50}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:46.863797 140972 storage/split_queue.go:116  [n1,split,s1,r15/1:/Table/1{8-9}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:47.864131 140972 storage/split_queue.go:116  [n1,split,s1,r16/1:/Table/{19-20}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:48.864450 140972 storage/split_queue.go:116  [n1,split,s1,r14/1:/Table/1{7-8}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:49.864743 140972 storage/split_queue.go:116  [n1,split,s1,r11/1:/Table/1{4-5}] shouldQueue: shouldQ=false priority=0.0
I180921 05:46:50.107319 140740 util/stop/stopper.go:537  quiescing; tasks left:
1      storage.intentResolver: processing intents
1      [async] closedts-rangefeed-subscriber
I180921 05:46:50.131373 140740 util/stop/stopper.go:537  quiescing; tasks left:
1      storage.intentResolver: processing intents
--- FAIL: TestSchemaChangeAfterCreateInTxn (6.31s)
	schema_changer_test.go:3289: pq: restart transaction: HandledRetryableTxnError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE): "sql txn" id=b7505e05 key=/Table/SystemConfigSpan/Start rw=true pri=0.00661602 iso=SERIALIZABLE stat=PENDING epo=0 ts=1537508804.861203547,2 orig=1537508804.277601919,0 max=1537508804.277601919,0 wto=false rop=false seq=16181

```

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -3234,6 +3234,18 @@ func TestSchemaChangeAfterCreateInTxn(t *testing.T) {
 	s, sqlDB, kvDB := serverutils.StartServer(t, params)
 	defer s.Stopper().Stop(context.TODO())
 
+	// The schema change below can occasionally take more than
+	// 5 seconds and gets pushed by the closed timestamp mechanism
+	// in the read timestamp cache. Setting the closed timestamp
+	// target duration to a higher value.
+	// TODO(vivek): Remove the need to do this by removing the use of
+	// txn.CommitTimestamp() in schema changes.
+	if _, err := sqlDB.Exec(`
+SET CLUSTER SETTING kv.closed_timestamp.target_duration = '20s'
+`); err != nil {
+		t.Fatal(err)
+	}
+
 	// A large enough value that the backfills run as part of the
 	// schema change run in many chunks.
 	var maxValue = 4001

